gradle/publish.gradle

```diff
@@ -1,9 +1,102 @@
+version = '0.3.1'
+String tag = "v$project.version"
+groovydoc.docTitle = 'Static Analysis Plugin'
+
 apply plugin: 'com.novoda.bintray-release'
 publish {
     userOrg = 'novoda'
     groupId = 'com.novoda'
     artifactId = 'gradle-static-analysis-plugin'
-    publishVersion = '0.3'
+    publishVersion = project.version
     website = 'https://github.com/novoda/gradle-static-analysis-plugin'
 }
 
+apply plugin: 'com.novoda.build-properties'
+buildProperties {
+    secrets {
+        file(rootProject.file('secrets.properties'), '''
+This file should contain:
+- git.username: the username used to push to the repo
+- git.password: the password used to push to the repo
+		''')
+    }
+}
+
+apply plugin: 'org.ajoberstar.grgit'
+apply plugin: 'org.ajoberstar.github-pages'
+
+githubPages {
+    commitMessage = "Deploy groovydoc for release $tag"
+    pages {
+        from groovydoc.destinationDir
+        into "docs/${project.version}"
+    }
+}
+
+task prepareGhCredentials {
+    description = 'Prepare GitHub credentials'
+    group = 'release'
+    doLast {
+        System.properties['org.ajoberstar.grgit.auth.username'] = buildProperties.secrets['git.username'].string
+        System.properties['org.ajoberstar.grgit.auth.password'] = buildProperties.secrets['git.password'].string
+    }
+}
+
+prepareGhPages.dependsOn groovydoc
+publishGhPages.dependsOn prepareGhCredentials
+
+task prepareRelease {
+    description = 'Prepare changelog and tag for release'
+    group = 'release'
+    dependsOn prepareGhPages, prepareGhCredentials
+    doLast {
+        String changelog = extractChangelog()
+        grgit.tag.add {
+            name = tag
+            message = "Release $tag\n\n$changelog"
+        }
+    }
+}
+
+String extractChangelog() {
+    String fullChangelog = rootProject.file('CHANGELOG.md').text
+    def latestChangelog = (fullChangelog =~ /Version ${project.version}.*\n-*([\s\S]*?)Version.*\n-*/)
```

Comment:
@ataulm yes, this is assuming the changelog is written always following a specific structure. The script here is a first draft, something I hacked together a while ago that I feel should not be just a stashed snippet. Soon a follow-up PR will enhance this script adding the missing piece of functionality (the creation of a release entry in GitHub) and the documentation on how to release (including the format of the changelog).